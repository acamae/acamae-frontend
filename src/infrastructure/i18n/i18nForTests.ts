import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

// Import locales
import enGB from './locales/en-GB.json';
import esES from './locales/es-ES.json';

// Configure i18n for tests
i18n.use(initReactI18next).init({
  lng: 'en-GB',
  fallbackLng: 'en-GB',
  debug: false,
  resources: {
    'en-GB': {
      translation: enGB,
    },
    'es-ES': {
      translation: esES,
    },
  },
  interpolation: {
    escapeValue: false,
  },
});

export default i18n;
