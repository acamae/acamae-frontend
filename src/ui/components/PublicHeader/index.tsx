import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';

import LanguageSelector from '@ui/components/LanguageSelector';

const PublicHeader: React.FC = () => {
  const { t } = useTranslation();

  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark px-3">
      <Link className="navbar-brand" to="/" data-testid="link-home">
        {t('app.name')}
      </Link>

      <div className="ms-auto d-flex align-items-center gap-3">
        <LanguageSelector />
        <Link to="/login" className="btn btn-outline-light btn-sm" data-testid="link-login-nav">
          {t('nav.login')}
        </Link>
        <Link to="/register" className="btn btn-primary btn-sm" data-testid="link-register">
          {t('nav.register')}
        </Link>
      </div>
    </nav>
  );
};

export default PublicHeader;
