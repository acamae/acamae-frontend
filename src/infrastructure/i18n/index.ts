/* eslint-disable import/no-named-as-default-member, import/no-named-as-default */
import i18n from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import { initReactI18next } from 'react-i18next';

import enGB from '@infrastructure/i18n/locales/en-GB.json';
import esES from '@infrastructure/i18n/locales/es-ES.json';

const resources = {
  'es-ES': { translation: esES },
  'en-GB': { translation: enGB },
};

i18n
  .use(LanguageDetector) // Detects language from browser
  .use(initReactI18next) // Integrates i18next with React
  .init({
    resources,
    fallbackLng: 'es-ES', // Default language if translation is not found
    supportedLngs: ['es-ES', 'en-GB'],
    interpolation: {
      escapeValue: false, // React already escapes values by default
    },
    detection: {
      order: ['localStorage', 'navigator'], // Detection order: first localStorage, then navigator
      caches: ['localStorage'], // Stores language preference in localStorage
    },
  });

export default i18n;
