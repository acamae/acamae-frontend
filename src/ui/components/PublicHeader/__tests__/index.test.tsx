import { render, screen } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter } from 'react-router-dom';

import i18n from '@infrastructure/i18n';
import es from '@infrastructure/i18n/locales/es-ES.json';
import PublicHeader from '@ui/components/PublicHeader';

jest.mock('@ui/components/LanguageSelector', () => {
  const MockLanguageSelector = () => <div data-testid="mock-language-selector"></div>;
  MockLanguageSelector.displayName = 'MockLanguageSelector';
  return MockLanguageSelector;
});

function renderPublicHeader() {
  return render(
    <I18nextProvider i18n={i18n}>
      <MemoryRouter>
        <PublicHeader />
      </MemoryRouter>
    </I18nextProvider>
  );
}

describe('PublicHeader', () => {
  it('debería renderizar el nombre de la app como un enlace a la home', () => {
    renderPublicHeader();
    const appNameLink = screen.getByTestId('link-home');
    expect(appNameLink).toBeInTheDocument();
    expect(appNameLink).toHaveTextContent(es.app.name);
    expect(appNameLink).toHaveAttribute('href', '/');
  });

  it('debería renderizar el LanguageSelector mockeado', () => {
    renderPublicHeader();
    expect(screen.getByTestId('mock-language-selector')).toBeInTheDocument();
  });

  it('debería renderizar los enlaces de login y registro correctamente', () => {
    renderPublicHeader();
    const loginLink = screen.getByTestId('link-login-nav');
    expect(loginLink).toBeInTheDocument();
    expect(loginLink).toHaveTextContent(es.nav.login);
    expect(loginLink).toHaveAttribute('href', '/login');

    const registerLink = screen.getByTestId('link-register');
    expect(registerLink).toBeInTheDocument();
    expect(registerLink).toHaveTextContent(es.nav.register);
    expect(registerLink).toHaveAttribute('href', '/register');
  });

  it('snapshot: renderizado de PublicHeader', () => {
    const { asFragment } = renderPublicHeader();
    expect(asFragment()).toMatchSnapshot();
  });
});
