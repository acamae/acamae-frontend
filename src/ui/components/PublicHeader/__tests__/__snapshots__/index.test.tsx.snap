// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicHeader snapshot: renderizado de PublicHeader 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark px-3"
  >
    <a
      class="navbar-brand"
      data-discover="true"
      data-testid="link-home"
      href="/"
    >
      Gestión eSports
    </a>
    <div
      class="ms-auto d-flex align-items-center gap-3"
    >
      <div
        data-testid="mock-language-selector"
      />
      <a
        class="btn btn-outline-light btn-sm"
        data-discover="true"
        data-testid="link-login-nav"
        href="/login"
      >
        Iniciar sesión
      </a>
      <a
        class="btn btn-primary btn-sm"
        data-discover="true"
        data-testid="link-register"
        href="/register"
      >
        Registrarse
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
